<launch>
	<arg name="gui" default="true" />
	<arg name="map_file" default="$(find robot_localization)/maps/willow_garage.yaml" />
	<param name="/use_sim_time" value="true" />

	<!-- load world -->
	<node name="gazebo" pkg="gazebo" type="gazebo" args="-u $(find robot_localization)/worlds/pr2_navigation.world" respawn="false" output="screen" />

	<!-- start gui -->
	<group if="$(arg gui)">
		<node name="gazebo_gui" pkg="gazebo" type="gui" respawn="false" output="screen" />
	</group>

	<!-- start up robot -->
	<include file="$(find pr2_gazebo)/launch/pr2.launch" />

	<!-- tuck arms -->
	<node name="tuckarm" pkg="pr2_tuckarm" type="tuck_arms.py" args="-l t -r t" />

	<!-- keyboard teleop -->
	<include file="$(find pr2_teleop)/teleop_keyboard.launch" />

	<!-- navigation -->
	<include file="$(find pr2_2dnav)/pr2_2dnav.launch" />
	<node pkg="rviz" type="rviz" name="$(anon rviz)" args="-d $(find robot_localization)/rviz/pr2_localization.rviz" />
	<node pkg="costmap_2d" type="costmap_2d_markers" name="voxel_visualizer">
		<remap from="voxel_grid" to="move_base_node/local_costmap/voxel_grid_throttled" />
	</node>
	<!-- <node name="snap_map_icp" pkg="SnapMapICP" type="SnapMapICP" respawn="false" output="screen" /> -->


	<!-- run the map server -->
	<node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)" />

	<!-- parameters that give you reasonable physics as well as good speed -->
	<!-- needed so that gripper joint stops on objects instead of oscillating forever -->
	<node name="sor_pgs_iters" pkg="dynamic_reconfigure" type="dynparam" args="set gazebo sor_pgs_iters 50" />
	<node name="sor_pgs_w" pkg="dynamic_reconfigure" type="dynparam" args="set gazebo sor_pgs_w 1.0" />
	<node name="time_step" pkg="dynamic_reconfigure" type="dynparam" args="set gazebo time_step 0.001" />
</launch>
